### ISSUE TITLE
Problems with signatures in generics

### ISSUE THREAD

**Issue by piklprado** [piLaboratory/sads#88]
Many of the methods for the generics radpred, octavpred and qqsad are issuing errors, e.g.

```r
> moths.ls <- fitls(moths)
> moths.ln <- fitlnorm(moths)
> moths.lnt <- fitlnorm(moths, trunc=0.5)
> moths.gs <- fitgs(moths)
> radpred(sad="ls", coef=as.list(coef(moths.ls)), S=length(moths), N=sum(moths), distr="D")
Error in radpred(sad = "ls", coef = as.list(coef(moths.ls)), S = length(moths),  : 
  argumento "trunc" ausente, sem padrão
> radpred(sad="lnorm", coef=as.list(coef(moths.ln)), S=length(moths), N=sum(moths), distr="C")
Error in radpred(sad = "lnorm", coef = as.list(coef(moths.ln)), S = length(moths),  : 
  argumento "trunc" ausente, sem padrão
> radpred(rad="gs", coef=as.list(coef(moths.gs)), S=length(moths), N=sum(moths))
Error in radpred(rad = "gs", coef = as.list(coef(moths.gs)), S = length(moths),  : 
  argumento "trunc" ausente, sem padrão
> octavpred(moths, sad="ls", coef=as.list(coef(moths.ls)))
Error in (function (classes, fdef, mtable)  (from sads-methods.R#350) : 
  unable to find an inherited method for function ‘octavpred’ for signature ‘"integer", "character", "missing", "list", "missing", "missing", "missing", "missing"’
> octavpred(moths, sad="lnorm", coef=as.list(coef(moths.ln)))
Error in (function (classes, fdef, mtable)  (from sads-methods.R#350) : 
  unable to find an inherited method for function ‘octavpred’ for signature ‘"integer", "character", "missing", "list", "missing", "missing", "missing", "missing"’
> octavpred(moths, sad="lnorm", coef=as.list(coef(moths.lnt)), trunc=0.5)
Error in (function (classes, fdef, mtable)  (from sads-methods.R#350) : 
  unable to find an inherited method for function ‘octavpred’ for signature ‘"integer", "character", "missing", "list", "numeric", "missing", "missing", "missing"’
> octavpred(moths, rad="gs", coef=as.list(coef(moths.gs)))
Error in (function (classes, fdef, mtable)  (from sads-methods.R#350) : 
  unable to find an inherited method for function ‘octavpred’ for signature ‘"integer", "missing", "character", "list", "missing", "missing", "missing", "missing"’
> octavpred(sad="ls", coef=as.list(coef(moths.ls)), S=length(moths), N=sum(moths))
Error in (function (classes, fdef, mtable)  (from sads-methods.R#350) : 
  unable to find an inherited method for function ‘octavpred’ for signature ‘"missing", "character", "missing", "list", "missing", "missing", "integer", "integer"’
> octavpred(sad="lnorm", coef=as.list(coef(moths.ln)), S=length(moths), N=sum(moths))
Error in (function (classes, fdef, mtable)  (from sads-methods.R#350) : 
  unable to find an inherited method for function ‘octavpred’ for signature ‘"missing", "character", "missing", "list", "missing", "missing", "integer", "integer"’
> octavpred(sad="lnorm", coef=as.list(coef(moths.lnt)), trunc=0.5, S=length(moths), N=sum(moths))
Error in (function (classes, fdef, mtable)  (from sads-methods.R#350) : 
  unable to find an inherited method for function ‘octavpred’ for signature ‘"missing", "character", "missing", "list", "numeric", "missing", "integer", "integer"’
> octavpred(rad="gs", coef=as.list(coef(moths.gs)), S=length(moths), N=sum(moths))
Error in (function (classes, fdef, mtable)  (from sads-methods.R#350) : 
  unable to find an inherited method for function ‘octavpred’ for signature ‘"missing", "missing", "character", "list", "missing", "missing", "integer", "integer"’
> qqsad(x=moths, sad="ls", coef=as.list(coef(moths.ls)), distr="D")
Error in (function (classes, fdef, mtable)  (from sads-methods.R#517) : 
  unable to find an inherited method for function ‘qqsad’ for signature ‘"integer", "character", "list", "missing", "character"’
> qqsad(x=moths, sad="lnorm", coef=as.list(coef(moths.ln)), distr="C")
Error in (function (classes, fdef, mtable)  (from sads-methods.R#517) : 
  unable to find an inherited method for function ‘qqsad’ for signature ‘"integer", "character", "list", "missing", "character"’
> qqsad(x=moths, sad="lnorm", coef=as.list(coef(moths.lnt)), trunc=0.5, distr="C")
Error in (function (classes, fdef, mtable)  (from sads-methods.R#517) : 
  unable to find an inherited method for function ‘qqsad’ for signature ‘"integer", "character", "list", "numeric", "character"’
 ```

Fixed the problem with missing `trunc` argument in `radpred ` while checking PR #86 (see commit 0a1ab4ec89533c400d0380adcb657aea9aff7e44 ). But no idea about the other problems. Code for the distr branch [here](https://github.com/piLaboratory/sads/files/15556/problem_with_generics.txt) and for the release and dev branch
[here](https://github.com/piLaboratory/sads/files/15663/problem_with_generics_round2.txt).

**Comment by piklprado** [comment]
More cues: most of the problems were already present in the last release and also in the dev branch:

```r
moths.ls <- fitls(moths)
moths.ln <- fitlnorm(moths)
moths.lnt <- fitlnorm(moths, trunc=0.5)
moths.gs <- fitgs(moths)

## Problems with argument trunc in current release
## Problems with argument trunc in dev branch
## ok in distr branch (probably because commit 0a1ab4e). Also works without argument distr
radpred(sad="ls", coef=as.list(coef(moths.ls)), S=length(moths), N=sum(moths), distr="D")
radpred(sad="lnorm", coef=as.list(coef(moths.ln)), S=length(moths), N=sum(moths), distr="C")
radpred(rad="gs", coef=as.list(coef(moths.gs)), S=length(moths), N=sum(moths))
## Ok in the current release (as trunc is provided)
## Ok in the dev branch
## Ok in distr branch
radpred(sad="lnorm", coef=as.list(coef(moths.lnt)), trunc=0.5, S=length(moths), N=sum(moths), distr="C")

## Signature problems in the current release
## Signature problems in the dev branch
## Signature problems in the distr branch
octavpred(moths, sad="ls", coef=as.list(coef(moths.ls)))
octavpred(moths, sad="lnorm", coef=as.list(coef(moths.ln)))
octavpred(moths, sad="lnorm", coef=as.list(coef(moths.lnt)), trunc=0.5)
octavpred(moths, rad="gs", coef=as.list(coef(moths.gs)))

octavpred(sad="ls", coef=as.list(coef(moths.ls)), S=length(moths), N=sum(moths))
octavpred(sad="lnorm", coef=as.list(coef(moths.ln)), S=length(moths), N=sum(moths))
octavpred(sad="lnorm", coef=as.list(coef(moths.lnt)), trunc=0.5, S=length(moths), N=sum(moths))
octavpred(rad="gs", coef=as.list(coef(moths.gs)), S=length(moths), N=sum(moths))

qqsad(x=moths, sad="ls", coef=as.list(coef(moths.ls)))
qqsad(x=moths, sad="lnorm", coef=as.list(coef(moths.ln)))
qqsad(x=moths, sad="lnorm", coef=as.list(coef(moths.lnt)), trunc=0.5)
```
But there are also problems with slots in the distr branch, see PR #86

**Comment by piklprado** [comment]
And one problem with logical test of trunc argument was solved by my commit 0a1ab4ec89533c400d0380adcb657aea9aff7e44 in branch distr

```r
## Problems with argument trunc in current release
## Problems with argument trunc in dev branch
## ok in distr branch (probably because commit 0a1ab4e). Also works without argument distr
radpred(sad="ls", coef=as.list(coef(moths.ls)), S=length(moths), N=sum(moths), distr="D")
radpred(sad="lnorm", coef=as.list(coef(moths.ln)), S=length(moths), N=sum(moths), distr="C")
radpred(rad="gs", coef=as.list(coef(moths.gs)), S=length(moths), N=sum(moths))
## Ok in the current release (as trunc is provided)
## Ok in the dev branch
## Ok in distr branch
radpred(sad="lnorm", coef=as.list(coef(moths.lnt)), trunc=0.5, S=length(moths), N=sum(moths), distr="C")

```

**Comment by andrechalom** [comment]
See #86 for details
